src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -407,7 +407,7 @@ static int dhcp_server_send_udp(sd_dhcp_server *server, be32_t destination,
                    rather than binding the socket. This will be mostly useful
                    when we gain support for arbitrary number of server addresses
                  */
-                pktinfo = (struct in_pktinfo*) CMSG_DATA(cmsg);
+                pktinfo = CMSG_TYPED_DATA(cmsg, struct in_pktinfo);
```

Comment:
Hm? I cannot follow. 

`CMSG_TYPED_DATA()` is mostly a typesafe version of `CMSG_DATA()` with alignment check. And no extra values are read from the cmsg header other than `CMSG_DATA()` reads. If it does not work, then I think `CMSG_DATA()` here cannot work.

> That's because it validates the size values in the cmsg header, but those are not initialized here.

Here "the size values" mean `cmsg->cmsg_len` ? It _is_ already initialized before `CMSG_TYPED_DATA()` (or `CMSG_DATA()`) is called.